### PR TITLE
Plot losses against elapsed training time via --x-axis flag

### PR DESCRIPTION
**Pull request by florianscheidl**
## Summary

Adds `--x-axis` support to `plot_train` so losses can be plotted against `elapsed_training_time_seconds` instead of samples.

## Changes

- **CLI**: Replace `--x_type` ('step'/'reltime') with `--x-axis` accepting column names `samples` (default) or `elapsed_training_time`
- **plot_loss_avg**: Added `x_axis` param (previously hardcoded `num_samples`); xlabel is auto-derived
- **Friendly labels**: When plotting against `elapsed_training_time_seconds`, x-axis labels read "elapsed training time [s]" instead of the raw column name
- All four plotting functions (`plot_lr`, `plot_loss_avg`, `plot_loss_per_stream`, `plot_loss_per_run`) now respect the `x_axis` param
- Removed dead `x_type` parameter from `plot_loss_per_stream`

## Usage

```bash
# Plot against samples (default)
python -m weathergen.utils.plot_training -fy runs.yml -o ./plots

# Plot against elapsed training time
python -m weathergen.utils.plot_training -fy runs.yml -o ./plots --x-axis elapsed_training_time
```

### CODE REVIEW

File: src/weathergen/run_train.py [hunk 1]
```diff
@@ -188,7 +189,7 @@ def run_train(args):
     trainer = Trainer(cf.train_logging)
 
     try:
-        trainer.run(cf, devices)
+        trainer.run(cf, devices, t_start=t_start)
```
Comment (clessig) [inline src/weathergen/run_train.py]: Can't we move this to Trainer?

File: config/streams/era5_georing/era5.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# (C) Copyright 2024 WeatherGenerator contributors.
```
Comment (clessig) [inline config/streams/era5_georing/era5.yml]: Remove from PR

File: config/streams/era5_georing/geos.yml [hunk 3]
```diff
@@ -0,0 +1,77 @@
+
```
Comment (clessig) [inline config/streams/era5_georing/geos.yml]: Remove from PR

File: config/config_era5_georing.yml [hunk 4]
```diff
@@ -0,0 +1,289 @@
+# (C) Copyright 2025 WeatherGenerator contributors.
```
Comment (clessig) [inline config/config_era5_georing.yml]: Remove from PR

File: src/weathergen/train/trainer.py [hunk 5]
```diff
@@ -378,7 +381,14 @@ def run(self, cf, devices, run_id_contd=None, mini_epoch_contd=None):
...
         self.validate_before_training()
 
+        # Log startup time
+        if is_root() and t_start is not None:
+            startup_time = time.time() - t_start
+            self.train_logger.log_metrics("train", {"startup_time_seconds": startup_time})
+            logger.info(f"Startup time: {startup_time:.2f} seconds")
+
         # training loop
+        self.t_training_start = time.time()
```
Comment (clessig) [inline src/weathergen/train/trainer.py:383]: For plot_train, timing should start here. This would also avoid that run_train is modified.

File: src/weathergen/train/trainer.py [hunk 6]
```diff
@@ -378,7 +381,14 @@ def run(self, cf, devices, run_id_contd=None, mini_epoch_contd=None):
         # run validation before training if requested
         self.validate_before_training()
 
+        # Log startup time
+        if is_root() and t_start is not None:
```
Comment (clessig) [inline src/weathergen/train/trainer.py]: I don't think this is needed.

File: src/weathergen/train/trainer.py [hunk 7]
```diff
@@ -561,6 +571,19 @@ def train(self, mini_epoch):
 
         self.dataset.advance()
 
+        if is_root():
+            total_training_time = time.time() - self.t_training_start
+            self.train_logger.log_metrics(
+                "train",
+                {
+                    "completed_mini_epoch": mini_epoch,
+                    "training_time_after_mini_epoch_seconds": total_training_time,
```
Comment (clessig) [inline src/weathergen/train/trainer.py]: Do we use this explicitly? Otherwise I would remove it. It can always be recovered by the time elapsed up to step k and the samples_per_mini_epoch (which is available through the config).

**Review by clessig — COMMENTED** [review]
Thanks for picking this up. It's important to keep Trainer as clean as possible so I would introduce only what is really necessary.

**Comment by florianscheidl** [comment]
> Thanks for picking this up. It's important to keep Trainer as clean as possible so I would introduce only what is really necessary.

Sorry, this should not have been a PR, only a commit. The correct PR is in https://github.com/ecmwf/WeatherGenerator/pull/2231

**Comment by clessig** [comment]
> > Thanks for picking this up. It's important to keep Trainer as clean as possible so I would introduce only what is really necessary.
> 
> Sorry, this should not have been a PR, only a commit. The correct PR is in #2231

But we should have plot_train with elapsed time on the x-axis and it should be a separate PR.
